bigchaindb/common/transaction.py

```diff
@@ -786,20 +786,14 @@ def to_inputs(self, condition_indices=None):
                 :obj:`list` of :class:`~bigchaindb.common.transaction.
                     Fulfillment`
         """
-        inputs = []
-        if condition_indices is None or len(condition_indices) == 0:
-            # NOTE: If no condition indices are passed, we just assume to
-            #       take all conditions as inputs.
-            condition_indices = [index for index, _
-                                 in enumerate(self.conditions)]
-
-        for cid in condition_indices:
-            input_cond = self.conditions[cid]
-            ffill = Fulfillment(input_cond.fulfillment,
-                                input_cond.owners_after,
-                                TransactionLink(self.id, cid))
-            inputs.append(ffill)
-        return inputs
+        # NOTE: If no condition indices are passed, we just assume to
+        #       take all conditions as inputs.
+        return [
+            Fulfillment(self.conditions[cid].fulfillment,
+                        self.conditions[cid].owners_after,
+                        TransactionLink(self.id, cid))
+            for cid in condition_indices or range(len(self.conditions))
```

Comment:
Because we only need the indices, not the values. `enumerate` is useful when iterating over both the the indices, and values.

If you look at the docs for [`enumerate`](https://docs.python.org/3/library/functions.html#enumerate), they show an equivalent of it:

```python
def enumerate(sequence, start=0):
    n = start
    for elem in sequence:
        yield n, elem
        n += 1
```

So, in the above code, we are only interested in `n`, not `elem`. Very often it is the other way around, such that one is only interested in `elem`.  

So if you only need `n`, `range(len(sequence))` will do. If you only need `elem`, then `for elem in sequence` will do. And if you need both: `for n, elem in enumerate(sequence)`.

Makes sense?